python/lsst/obs/base/formatters/packages.py

```diff
@@ -120,5 +125,6 @@ def _toBytes(self, inMemoryDataset):
         Exception
             The object could not be serialized.
         """
-        format = "yaml" if self.extension == ".yaml" else "pickle"
+        format_map = {".yaml": "yaml", ".pickle": "pickle", ".pkl": "pickle", ".json": "json"}
+        format = format_map.get(self.extension, "yaml")
```

Comment:
```suggestion
        format = self.extension.lstrip(".")
```

Reply:
In theory it could be a `.pkl` file but I guess we can declare that the package formatter is never going to do that so it doesn't have to know about such an extension.